catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -311,6 +340,74 @@ private Boolean userSpellcheckIsOn(QueryRequest request) {
     return userSpellcheckChoice;
   }
 
+  private Boolean userHighlightIsOn(QueryRequest request) {
+    Boolean userHighlight = false;
+    if (request.getProperties().get(HIGHLIGHT_KEY) != null) {
+      userHighlight = (Boolean) request.getProperties().get(HIGHLIGHT_KEY);
+    }
+    return userHighlight;
+  }
+
+  private void enableHighlighter(SolrQuery query) {
+    query.setParam(SOLR_HIGHLIGHT_KEY, true);
+    query.setParam("hl.fl", "*");
+    query.setParam("hl.simple.pre", "<b>");
+    query.setParam("hl.tag.pre", "<b>");
+    query.setParam("hl.simple.post", "</b>");
+    query.setParam("hl.tag.post", "</b>");
+    query.setParam("hl.requireFieldMatch", true);
+    query.setParam("hl.method", "unified");
+  }
+
+  private void extractHighlighing(QueryResponse response, Map<String, Serializable> responseProps) {
+    Map<String, Map<String, List<String>>> highlights = response.getHighlighting();
+    if (highlights == null) {
+      return;
+    }
+    Map<String, Map<String, Set<String>>> resultsHighlights = new HashMap<>();
+    for (String metacardId : highlights.keySet()) {
+      Map<String, List<String>> fieldHighlight = highlights.get(metacardId);
+      Map<String, Set<String>> consolidated = new HashMap<>();
+      if (fieldHighlight != null && fieldHighlight.size() > 0) {
+        for (Map.Entry<String, List<String>> entry : fieldHighlight.entrySet()) {
+          String solrField = entry.getKey();
+          String normalizedKey = resolver.resolveFieldName(solrField);
+          if (isHighlightBlacklisted(normalizedKey) || isHighlightBlacklisted(solrField)) {
```

Comment:
:pencil2: I would also suggest hiding any fields that are not part of the metacard type fields.  That will hide fields like `metacard_type_name_txt`, `metacard_type_obj`, and `metacard_source_txt`.

Reply:
The `isHighlightBlacklisted` method checks the `DynamicSchemaResolver.isPrivateField()` method as well so most of those are filtered out by that.  Oddly, `metacard_source_txt` is not in that list,, nor are the `metadata` fields.  I could add those to the private fields list in the resolver or to a different internal blacklist if that makes more sense.

Reply:
It looks like the metacard type fields are in that list.

https://github.com/codice/ddf/blob/55ce87a/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java#L116-L117

Metacard source isn't since it does not know about it.  The SolrCache in the standard framework adds that when using its composed SolrMetacardClient.

https://github.com/codice/ddf/blob/55ce87a/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SolrCache.java#L76-L83

Reply:
I reworked this section a little to use the specific metacard type for the result being highlighted.